lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -304,6 +239,90 @@ class FontSize extends Gatherer {
     return {analyzedFailingNodesData, analyzedFailingTextLength};
   }
 
+  /**
+   * @param {LH.Crdp.DOMSnapshot.CaptureSnapshotResponse} snapshot
+   * @return {BackendIdsToPartialFontData}
+   */
+  calculateBackendIdsToPartialFontData(snapshot) {
+    // Makes the strings access code easier to read.
+    /** @param {number} index */
+    const lookup = (index) => snapshot.strings[index];
+
+    // The document under analysis is the root document.
+    const doc = snapshot.documents[0];
+
+    // doc is a flattened property list describing all the Nodes in a document, with all string values
+    // deduped in a strings array.
+    // Implementation:
+    // https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/inspector/inspector_dom_snapshot_agent.cc?sq=package:chromium&g=0&l=534
+
+    if (!doc || !doc.nodes.nodeType || !doc.nodes.nodeName || !doc.nodes.backendNodeId
+      || !doc.nodes.nodeValue || !doc.nodes.parentIndex) {
+      throw new Error('Unexpected response from DOMSnapshot.captureSnapshot.');
+    }
+
+    // Not all nodes have computed styles (ex: TextNodes), so doc.layout.* is smaller than doc.nodes.*
+    // doc.layout.nodeIndex maps the index into doc.nodes.* to an index into doc.layout.styles.
+    // nodeIndexToStyleIndex inverses that mapping.
+    /** @type {Map<number, number>} */
+    const nodeIndexToStyleIndex = new Map();
+    for (let i = 0; i < doc.layout.nodeIndex.length; i++) {
+      nodeIndexToStyleIndex.set(doc.layout.nodeIndex[i], i);
+    }
+
+    /** @type {BackendIdsToPartialFontData} */
+    const backendIdsToPartialFontData = new Map();
+    for (let i = 0; i < doc.nodes.nodeType.length; i++) {
+      const nodeType = doc.nodes.nodeType[i];
+      const nodeValue = lookup(doc.nodes.nodeValue[i]);
+      if (!isTextNode({
+        nodeType,
+        parentNodeName: lookup(doc.nodes.nodeName[doc.nodes.parentIndex[i]]),
+      })) continue;
+
+      const styleIndex = nodeIndexToStyleIndex.get(doc.nodes.parentIndex[i]);
+      if (!styleIndex) continue;
+      const textLength = getNodeTextLength(nodeValue);
+      if (!textLength) continue; // ignore empty TextNodes
+      const parentStyles = doc.layout.styles[styleIndex];
+      const [fontSizeStringId] = parentStyles;
+      const fontSize = parseInt(lookup(fontSizeStringId), 10);
```

Comment:
maybe another comment here that we expect strings like `'16px'`, hence why the `parseInt` is appropriate

Reply:
also, do we actually want `parseFloat`? I guess 11.5 will still fail the < 12 check, but we might as well store the most accurate info